### ISSUE TITLE
[🚀 Feature]: Support disable-build-check in c# chromedriver implementation

### ISSUE THREAD

**Issue by ojintoad** [SeleniumHQ/selenium#10898]
### Feature and motivation

Sometimes Chrome/Chromedriver versions that are compatible have bugs which makes upgrading or downgrading temporarily to a distinct chromedriver version desirable.

One example is:

https://bugs.chromium.org/p/chromedriver/issues/detail?id=4121

This was fixed in the next beta release of chromedriver for several weeks before getting backported to the version where it was originally broke. Trying to run the beta 104 chromedriver against the stable 103 chrome browser was impossible in C# powered selenium automation because the current ChromeDriverService and ChromiumService in C#:

* Uses sealed classes and protected constructors
* Doesn't have support for the property `disableBuildCheck` arguments

For reference, this is the flag in chromedriver:

https://github.com/bayandin/chromedriver/blob/d9865ee18a2ac228d4e5f41ff2c950992cc0b788/chrome_launcher.cc#L280

Java supports the flag, as introduced by @diemol:

https://github.com/SeleniumHQ/selenium/blob/c7be1be9e1053488d13aab76a76598dbf0a39a56/java/src/org/openqa/selenium/chrome/ChromeDriverService.java#L274-L276

C# does not:

https://github.com/SeleniumHQ/selenium/blob/c7be1be9e1053488d13aab76a76598dbf0a39a56/dotnet/src/webdriver/Chromium/ChromiumDriverService.cs#L123

Can this be added to the c# implementation of chromedriver?

### Usage example

I tried to explain that in the Feature and Motivation, let me know if that wasn't clear enough from that section and I can attempt to explain further.

**Comment by titusfortner** [comment]
This line in the Java code is a shortcut that is used by the grid, and isn't necessary for the functionality you want.

In C# you just need to do:

```csharp
ChromeOptions options = new ChromeOptions();
options.AddArgument("--disable-build-check");
```

**Comment by ojintoad** [comment]
If that's true, that will be a relief. 

My understanding was that ChromeOptions are not passed to the ChromeDriver as arguments but rather to the ChromeBrowser process it starts up. It sounds like you're saying I'm mistaken on that.

I will work with my team to confirm your example code works today.

**Comment by ojintoad** [comment]
Just to be clear, my understanding was based on this post ["--disable-build-check has no effect"](https://groups.google.com/g/chromedriver-users/c/1GffkCgwgqQ) - admittedly it is old (2019) and for Java, and C# may behave differently.

> This flag isn't a Chrome switch, it's a ChromeDriver switch. The Selenium libraries don't make these easily accessible to code. If you start ChromeDriver independently on the command line, then of course it is easy to add the switch. At the bottom of this page are instructions for connecting to such an instance: https://chromedriver.chromium.org/getting-started

**Comment by ojintoad** [comment]
@titusfortner I tested locally and the code you've suggested doesn't seem to work.

Steps I took:

1. Updated my project to use the preprelease nuget chromedriver package of https://www.nuget.org/packages/Selenium.WebDriver.ChromeDriver/104.0.5112.2900-beta
2. Added the code before the driver gets created like...
```
options.AddUserProfilePreference("permissions.default.microphone", 1);
options.AddArgument("--disable-build-check");

var driver = new ChromeDriver(driverService, options);
```
3. Ran a Selenium test. The exception I got back was:

> System.InvalidOperationException: 'session not created: This version of ChromeDriver only supports Chrome version 104
Current browser version is 103.0.5060.134 with binary path C:\Program Files (x86)\Google\Chrome\Application\chrome.exe (SessionNotCreated)'

Just to cover an odd case, I also tried 

`options.AddArgument("disable-build-check");`

without the prefix dashes, and that also did not work.

I should have clarified as well in the original opening of this, I had gotten clearance to write up a feature request from @diemol in the Selenium users slack after discussing there. I'll tag you into that thread.

Is there anything else you would want me to try? If not, can we please reopen this?

**Comment by titusfortner** [comment]
Oh, yeah, I see what you mean, this is an arg for the driver, not the browser. Weird that .NET and Java lock this down so much, but sure.

Fixed: 6f082ed6c3dd60c498ecec00ab20392acf12748f

Available next release
